File: src/main/scala/mffs/mobilize/event/BlockNotifyDelayedEvent.java
package mffs.mobilize.event;

import cpw.mods.fml.common.Loader;
import cpw.mods.fml.relauncher.ReflectionHelper;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import universalelectricity.core.transform.vector.Vector3;

/**
 * Removes the TileEntity
 *
 * @author Calclavia
 */
public class BlockNotifyDelayedEvent extends DelayedEvent
{
	private World world;
	private Vector3 position;

	public BlockNotifyDelayedEvent(IDelayedEventHandler handler, int ticks, World world, Vector3 position)
	{
		super(handler, ticks);
		this.world = world;
		this.position = position;
	}

	@Override
	protected void onEvent()
	{
		if (!this.world.isRemote)
		{
			this.world.notifyBlocksOfNeighborChange(this.position.xi(), this.position.yi(), this.position.zi(), this.position.getBlockID(this.world));

			TileEntity newTile = this.position.getTileEntity(this.world);

			if (newTile != null)
			{
				if (Loader.isModLoaded("BuildCraft|Factory"))
				{
					/**
					 * Special quarry compatibility code.
					 */
					try
					{
						Class clazz = Class.forName("buildcraft.factory.TileQuarry");

						if (clazz == newTile.getClass())
						{
							ReflectionHelper.setPrivateValue(clazz, newTile, true, "isAlive");
						}
					}
					catch (Exception e)
					{
						e.printStackTrace();
					}
				}
			}
		}
	}
}

File: src/main/scala/mffs/field/mobilize/event/BlockNotifyDelayedEvent.scala
package mffs.field.mobilize.event

import mffs.field.mobilize.BlockMobilizer

/**
 * Removes the TileEntity
 *
 * @author Calclavia
 */
class BlockNotifyDelayedEvent(handler: IDelayedEventHandler, ticks: Int, world: World, position: Vector3d) extends DelayedEvent(handler, ticks)
{
  protected override def onEvent
  {
    if (!this.world.isRemote)
    {
      world.notifyBlocksOfNeighborChange(position.xi, position.yi, position.zi, position.getBlock(world))
      val newTile = position.getTileEntity(world)

      if (newTile != null)
      {
        if (Loader.isModLoaded("BuildCraft|Factory"))
        {
          try
          {
            val clazz = Class.forName("buildcraft.factory.TileQuarry").asInstanceOf[Class[_ >: Any]]

            if (clazz == newTile.getClass)
            {
              ReflectionHelper.setPrivateValue(clazz, newTile, true, "isAlive")
            }
          }
          catch
            {
              case e: Exception =>
              {
                e.printStackTrace
              }
            }
        }

		  handler.asInstanceOf[BlockMobilizer].performingMove = false
      }
    }
  }
}